### PR TITLE
Add TaskConnectionHandler and TaskHttpConnectionHandler

### PR DESCRIPTION
**Pull request by david-eckardt-sociomantic**
`TaskConnectionHandler` and `TaskHttpConnectionHandler` are the task based equivalents to `IFiberConnectionHandler` and `HttpConnectionHandler`.

### CODE REVIEW

File: src/ocean/net/server/connection/TaskConnectionHandler.d [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    Imports
+
+*******************************************************************************/
+
+import ocean.net.server.connection.IConnectionHandler;
+import ocean.util.container.pool.model.IResettable;
+
+/*******************************************************************************
+
+    Connection handler abstract base class.
```
Comment (gavin-norman-sociomantic) [inline src/ocean/net/server/connection/TaskConnectionHandler.d]: If you removed the `/* Imports */` comment, you could use `/// ditto` here.

File: test/httpserver/main.d [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    Test-suite for the task based HTTP server.
+
+    This test uses a TCP socket connection to `localhost:8080`.
+
+    FLAKY: the unittests in this module are a bit flaky, as they rely on making
+    various system calls (`socket`, `connect`, `read/write`, epoll API
+    functions, etc) which could, under certain environmental conditions, fail.
+
+    Copyright:      Copyright (c) 2017 sociomantic labs. All rights reserved
+
```
Comment (gavin-norman-sociomantic) [inline test/httpserver/main.d:12]: License missing.

File: src/ocean/net/http/TaskHttpConnectionHandler.d [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+
+ ******************************************************************************/
+
+import ocean.transition;
+
+import ocean.net.server.connection.TaskConnectionHandler;
+
+/******************************************************************************/
+
+abstract class TaskHttpConnectionHandler : TaskConnectionHandler
```
Comment (gavin-norman-sociomantic) [inline src/ocean/net/http/TaskHttpConnectionHandler.d]: Same comment about `/* Imports */` and `/// ditto`.

File: src/ocean/net/http/TaskHttpConnectionHandler.d [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+    import ocean.io.select.protocol.generic.ErrnoIOException: IOError;
+
+    import ocean.sys.socket.AddressIPSocket;
+    import ocean.sys.ErrnoException;
+    import ocean.core.Enforce;
+
+
+    /**************************************************************************
+
+        HTTP request message parser and response message generator
```
Comment (gavin-norman-sociomantic) [inline src/ocean/net/http/TaskHttpConnectionHandler.d]: A small point, but I don't like "X and Y" comments like this. It's clearer to have a separate comment for each member.

File: src/ocean/net/http/TaskHttpConnectionHandler.d [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+    protected HttpRequest   request;
+    protected HttpResponse  response;
+
+    /**************************************************************************
+
+        Reused exception instance; may be thrown by a subclass as well.
+
+     **************************************************************************/
+
+    protected HttpException             http_exception;
```
Comment (gavin-norman-sociomantic) [inline src/ocean/net/http/TaskHttpConnectionHandler.d]: Apparently no purpose for the extra indentation.

Comment (gavin-norman-sociomantic) [reply src/ocean/net/http/TaskHttpConnectionHandler.d]: (Not reviewing the rest of the code in this module as I presume it's just copied from the existing HTTP connection handler.)

**Comment by mihails-strasuns-sociomantic** [comment]
Ping @david-eckardt-sociomantic :)

**Comment by david-eckardt-sociomantic** [comment]
Updated, addressing all comments and adding a warning to the release notes that `Task(Http)ConnectionHandler` hasn't been tested for performance in a production environment yet.

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Comment by mihails-strasuns-sociomantic** [comment]
OK, let's do it. I will probably have more detailed look myself next time I get to working on task features and optimization as it looks like great source of practical cases.

**Pull request merged**